### DISCUSSION TITLE
Support for FT232HPQ  and UMFT232HPEV-S

### DISCUSSION THREAD

**Discussion by mdahlkilt** [eblot/pyftdi#371] [Q&A]
I recently bought 
USB Hi-Speed FT232HP Development Module
https://ftdichip.com/products/umft232hpev-s/

I was expecting to be able to use this together with pyFtdi - but I can't get it to work 
I keep getting the "No USB-Serial device has been detected"

Here is a list of observations:
ftdi_urls.py returns an empty result.

Using lsusb I can see the device id:
Bus 005 Device 013: ID 0403:6045 Future Technology Devices International, Ltd Single RS232-HS
The kernel detect the USB device and uses the ftdi_sio module

dmesg reports:
[623554.215489] ftdi_sio 5-3.4:1.0: FTDI USB Serial Device converter detected
[623554.215520] usb 5-3.4: Detected FT232HP
[623554.216152] usb 5-3.4: FTDI USB Serial Device converter now attached to ttyUSB0

I've updated the udev rules:
```
# /etc/udev/rules.d/11-ftdi.rules

# FT232AM/FT232BM/FT232R
SUBSYSTEM=="usb", ATTR{idVendor}=="0403", ATTR{idProduct}=="6001", GROUP="plugdev", MODE="0664"
# FT2232C/FT2232D/FT2232H
SUBSYSTEM=="usb", ATTR{idVendor}=="0403", ATTR{idProduct}=="6010", GROUP="plugdev", MODE="0664"
# FT4232/FT4232H
SUBSYSTEM=="usb", ATTR{idVendor}=="0403", ATTR{idProduct}=="6011", GROUP="plugdev", MODE="0664"
# FT232H
SUBSYSTEM=="usb", ATTR{idVendor}=="0403", ATTR{idProduct}=="6014", GROUP="plugdev", MODE="0664"
# FT230X/FT231X/FT234X
SUBSYSTEM=="usb", ATTR{idVendor}=="0403", ATTR{idProduct}=="6015", GROUP="plugdev", MODE="0664"
# FT232HS
SUBSYSTEM=="usb", ATTR{idVendor}=="0403", ATTR{idProduct}=="6045", GROUP="plugdev", MODE="0664"
# FT4232HA
SUBSYSTEM=="usb", ATTR{idVendor}=="0403", ATTR{idProduct}=="6048", GROUP="plugdev", MODE="0664"
```
Any ideas what is wrong?

**Comment by eblot** [comment]
Hi.

FT232HPQ is not yet supported as I do not have any HW to test it. I need to find one, but FTDI chip fees are just ridiculous...

**Reply by eblot** [reply]
Is this board compatible https://eu.mouser.com/ProductDetail/FTDI/UMFT232HPEV-S?qs=doiCPypUmgFim41DTjQwSg%3D%3D
(sorry, in a hurry...)

**Reply by mdahlkilt** [reply]
I got it working locally, with the following patch
[pyftdi-ft232hp.patch.gz](https://github.com/eblot/pyftdi/files/14994306/pyftdi-ft232hp.patch.gz)
(Sorry for the gzip, but the attach file will not accept my patch file)

**Comment by pedroluisneves** [comment]
Hi @eblot. As this device is out for some time, and we seem to have a quick way to add support to it thanks to @mdahlkilt, so there is any possibility to integrated the suggested changes to the lib?

(latest ftdi.py with the changes)
[ftdi.py](https://github.com/user-attachments/files/31703372/ftdi.py)

**Comment by eblot** [comment]
Hey @pedroluisneves,

Someone needs to create a PR, with updated tests and documentation, and the CI should pass.
If everything is ok, yes it should be possible.

**Reply by pedroluisneves** [reply]
Hi @eblot, to be honest I never made any pull request in the past, therefore if you could support me on this it would be great.
I hade the chance to test the suggested changes from @mdahlkilt, and it works perfectly with my I2C application.
